app/upgrades/v8/forks.go

```diff
@@ -0,0 +1,41 @@
+package v8
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	superfluidkeeper "github.com/osmosis-labs/osmosis/v7/x/superfluid/keeper"
+)
+
+// RunForkLogic executes height-gated on-chain fork logic for the Osmosis v3
+// upgrade.
+func RunForkLogic(ctx sdk.Context, superfluid *superfluidkeeper.Keeper) {
+	ctx.Logger().Info("Applying Osmosis vx upgrade." +
+		"Allowing direct unbonding for whitelisted pools")
+	RegisterWhitelistedDirectUnbondPools(ctx, superfluid)
+}
+
+// RegisterWhitelistedDirectUnbondPools registers pools that are allowed to unpool
+// https://www.mintscan.io/osmosis/proposals/226
+// osmosisd q gov proposal 226
+func RegisterWhitelistedDirectUnbondPools(ctx sdk.Context, superfluid *superfluidkeeper.Keeper) {
+	// These are the pools listed in the proposal. Proposal raw text for the listing of UST pools:
+	// 	The list of pools affected is:
+	// #560 (UST/OSMO)
+	// #562 (UST/LUNA)
+	// #567 (UST/EEUR)
+	// #578 (UST/XKI)
+	// #592 (UST/BTSG)
+	// #610 (UST/CMDX)
+	// #612 (UST/XPRT)
+	// #615 (UST/LUM)
+	// #642 (UST/UMEE)
+	// #679 (4Pool)
+	whitelistedPoolShares := []int64{560, 562, 567, 578, 592, 610, 612, 615, 642, 679}
```

Comment:
Though the poolIDs all look correct at present, Is there some way to provide additional verification of the pool pairs?

For instance verify that 

560 is  LUNA/osmo
`ibc/BE1BB42D4BE3C30D50B68D7C41DB4DFCE9678E8EF8C539F6E6A9345048894FCC`
`uosmo`

562 is UST/LUNA
`ibc/0EF15DF2F02480ADE0BB6E85D9EBB5DAEA2836D3860E9F97F9AADE4F57A31AA0`
`ibc/BE1BB42D4BE3C30D50B68D7C41DB4DFCE9678E8EF8C539F6E6A9345048894FCC`

(just speaking from recent foot gun experience)

Reply:
good call. We can add a function for "Check pool contains UST", and then do `gammKeeper.GetPool(poolID)`, and then iterate through that pools balances, and see if anyone is UST.

Do you have bandwidth to make a PR here for this?

Reply:
Also had a mini-panic there, 560 is UST/Osmo :sweat_smile:

Reply:
Sorry, I tried and tried to make sure I didn't say that

Reply:
https://github.com/osmosis-labs/osmosis/pull/1482 implements "CheckPoolContainsUST"

Reply:
good job, joe

Reply:
sweet ty!